library/src/scala/tasty/TypeTest.scala

```diff
@@ -0,0 +1,27 @@
+package scala.tasty
+
+/** A `TypeTest[S, T]` (where `T <: S`) contains the logic needed to know at runtime if a value of
+ *  type `S` can be downcased to `T`.
+ *
+ *  If a pattern match is performed on a term of type `s: S` that is uncheckable with `s.isInstanceOf[T]` and
+ *  the pattern are of the form:
+ *    - `t: T`
+ *    - `t @ X()` where the `X.unapply` has takes an argument of type `T`
+ *  then a given instance of `TypeTest[S, T]` is summoned and used to performed the test.
+ *
+ *  Note: This is replacemet for `ClassTag.unapply` that can be sound for path dependent types
+ */
+@scala.annotation.implicitNotFound(msg = "No TypeTest available for [${S}, ${T}]")
+trait TypeTest[S, T <: S] extends Serializable {
```

Comment:
I don't think we should enforce `T <: S`. With regular type test I can do:
```scala
scala> val x: CharSequence = "hello"
x: CharSequence = hello

scala> x match {
     |   case y: Comparable[_] => "ok"
     |   case _ => "ko"
     | }
res0: String = ok
```
soundly and without warning, even though `CharSequence` and `Comparable` are unrelated.

Reply:
Removed `T <: S`